tests/test-list-fallback.py

```diff
@@ -62,3 +62,38 @@
 @with_connection
 async def test_client_list_override_invalid_raw_command(loop, client, server):
     await client.list(raw_command="FOO")
+
+def test_client_list_windows():
+    p = aioftp.Client.parse_list_line_windows
+    test_str = """
+ 11/4/2018   9:09 PM  <DIR>         .
+ 8/10/2018   1:02 PM  <DIR>         ..
+ 9/23/2018   2:16 PM  <DIR>         bin
+10/16/2018  10:25 PM  <DIR>         Desktop
+ 11/4/2018   3:31 PM  <DIR>         dow
+10/16/2018   8:21 PM  <DIR>         Downloads
+10/14/2018   5:34 PM  <DIR>         msc
+  9/9/2018   9:32 AM  <DIR>         opt
+ 10/3/2018   2:58 PM    34,359,738,368  win10.img
+ 6/30/2018   8:36 AM    3,939,237,888  win10.iso
+ 7/26/2018   1:11 PM           189  win10.sh
+10/29/2018  11:46 AM    34,359,738,368  win7.img
+ 6/30/2018   8:35 AM    3,319,791,616  win7.iso
+10/29/2018  10:55 AM           219  win7.sh   
+       6 files           75,978,506,648 bytes
+       3 directories     22,198,362,112 bytes free
+  """
+    test_str = test_str.strip().split("\n")
+    entities = []
+    for x in test_str:
+        try:
+            entities.append(p(aioftp.Client,x))
+        except ValueError as e:
+            pass
```

Comment:
Don't get this part. Why do you catch `ValueError`? Is it for empty lines? If so filter them with `if`.

Reply:
Not all lines outputted by `dir` are useful. There is a banner at the beginning and there is some trailing data at the end that is not useful for our purposes. We use `async` and I haven't found a good way to ignore these lines specifically and I'm not sure if the format of those useless lines are the same across versions of `dir`. We also can't be sure if the target server is Unix or Windows until we try to parse a line, meaning we can't simply ignore the first K lines or so on. Currently exceptions are used to indicate that the line can't be parsed.

TL; DR: due to the nature of `async`, `ValueError` is used to ignore the `dir` banners and footer.

Reply:
Better approach is `contextlib.suppress` in this case, I think.